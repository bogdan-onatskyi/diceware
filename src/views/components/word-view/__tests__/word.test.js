import React from 'react';

import Word from '../word';

describe("Word component", () => {
    it("should render a word with tooltip", () => {
        const wrapper = shallow(
            <Word mod="prev1" word="word" tip="Выбрать это слово"/>
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('div').prop('data-tip')).toEqual("Выбрать это слово");
    });

    it("should render a word without tooltip", () => {
        const wrapper = shallow(
            <Word mod="prev1" word="" tip="Выбрать это слово"/>
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('div').prop('data-tip')).toEqual("");
    });

    it("should have 4 props", () => {
        const propsStr = 'className mod word tip';
        const props = [];
        for (let prop in Word.propTypes) props.push(prop);
        expect(props.join(' ') === propsStr).toBe(true);
    });
});
