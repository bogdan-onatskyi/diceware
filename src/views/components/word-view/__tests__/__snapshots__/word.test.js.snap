// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Word component should render a word with tooltip 1`] = `
<div
  className="wv__word wv__word--prev1"
  data-tip="Выбрать это слово"
>
   
  word
   
</div>
`;

exports[`Word component should render a word without tooltip 1`] = `
<div
  className="wv__word wv__word--prev1 wv__word--disabled"
  data-tip=""
>
   
   
</div>
`;
